plugins/CoreAdminHome/OptOutManager.php

```diff
@@ -158,10 +181,333 @@ public function getQueryParameters()
+        $useSecureCookies = Common::getRequestVar('useSecureCookies', 1, 'int');
+
+        // If the language parameter is 'auto' then use the browser language
+        if ($language === 'auto') {
+            $language = Common::extractLanguageCodeFromBrowserLanguage(Common::getBrowserLanguage(),
+                                       APILanguagesManager::getInstance()->getAvailableLanguages());
+        }
+
+        $settings = [
+            'showIntro' => $showIntro,
+            'divId' => $divId,
+            'useSecureCookies' => $useSecureCookies,
+            'cookiePath' => TrackerConfig::getConfigValue('cookie_path'),
+            'cookieDomain' => TrackerConfig::getConfigValue('cookie_domain'),
+            'cookieSameSite' => 'Lax',
+            'useCookiesIfNoTracker' => $useCookiesIfNoTracker,
+            'useCookiesTimeout' => $useCookiesTimeout,
+        ];
+
+        // Self contained code translations are static and always use the language of the user who generated the embed code
+        $translations = $this->getTranslations($language);
+        $translations['OptOutErrorNoTracker'] = Piwik::translate('CoreAdminHome_OptOutErrorNoTracker', [], $language);
+        $settings = array_merge($settings, $translations);
+        $settingsString = 'var settings = '.json_encode($settings).';';
+
+        $styleSheet = $this->optOutStyling(null, null, null, null, true);
+
+        /** @lang JavaScript */
+        $code = <<<JS
+
+        var settings = {};          
+        var checkForTrackerTried = 0;
+        var checkForTrackerTries = (settings.useCookiesTimeout * 4);
+        var checkForTrackerInterval = 250;
+        var optOutDiv = null;
+        
+        function optOutInit() {
+            optOutDiv = document.getElementById(settings.divId);
+            if (optOutDiv) {
+                optOutDiv.style.cssText += 'stylecss'; // Appending css to avoid overwritting existing inline div styles
+            } else {
+                showContent(false, null, true); // will show unable to find opt-out div error
+                return;                
+            }
+            checkForMatomoTracker();
+        }
+        
+        function checkForMatomoTracker() {
+            if (typeof _paq === 'undefined') {                
+                if (checkForTrackerTried < checkForTrackerTries) {
+                    setTimeout(checkForMatomoTracker, checkForTrackerInterval);
+                    checkForTrackerTried++;
+                    return;
+                } else {
+                    if (settings.useCookiesIfNoTracker) {
+                        showOptOutDirect();
+                        return;
+                    } else {
+                        console.log('Matomo OptOutJS: failed to find Matomo tracker after '+(checkForTrackerTries*checkForTrackerInterval / 1000)+' seconds');
+                        return;    
+                    }
+                }
+            }
+            showOptOutTracker();
```

Comment:
Guess this could be written a bit simpler like:
```suggestion
            if (typeof _paq !== 'undefined') {
                showOptOutTracker();
                return;
            }
            
            if (checkForTrackerTried < checkForTrackerTries) {
                setTimeout(checkForMatomoTracker, checkForTrackerInterval);
                checkForTrackerTried++;
                return;
            }
            
            if (settings.useCookiesIfNoTracker) {
                showOptOutDirect();
                return;
            }
            
            console.log('Matomo OptOutJS: failed to find Matomo tracker after '+(checkForTrackerTries*checkForTrackerInterval / 1000)+' seconds');
```